### PR TITLE
ci: optimize build

### PR DESCRIPTION
**Pull request by splincode**
## PR Checklist

Please check if your PR fulfills the following requirements:

- [x] The commit message follows [Conventional Commits](https://www.conventionalcommits.org/en/)
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

## PR Type

What kind of change does this PR introduce?

- [ ] Bugfix
- [ ] Feature
- [ ] Refactoring
- [ ] Code style update
- [x] Build or CI related changes
- [ ] Documentation content changes

## What is the current behavior?

image

image

Build time: 130 sec

## What is the new behavior?

image

image

Decompose 2.29Mb into 2 files
Build time: 97 sec

## Does this PR introduce a breaking change?

- [ ] Yes
- [x] No

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -21,7 +21,7 @@
         "stylelint": "stylelint '**/*.less'",
         "lint": "eslint --cache --cache-location node_modules/.cache/eslint",
         "prettier": "prettier '**/*.{svg,yml,js,ts,html,md,less,json}' --write",
-        "typecheck": "tsc --noEmit --skipLibCheck --project tsconfig.spec.json",
```
Comment (waterplea) [inline package.json:24]: Why this change?

File: tsconfig.json [hunk 2]
```diff
@@ -34,6 +34,18 @@
         "noUnusedLocals": true,
         "noImplicitReturns": true,
         "noUnusedParameters": true,
+        "incremental": true,
+        "newLine": "lf",
+        "isolatedModules": false,
+        "emitDecoratorMetadata": true,
```
Comment (waterplea) [inline tsconfig.json:40]: Why do we need decorator metadata?

File: package.json (hunk 1 above)
Comment (splincode) [reply package.json:24]: because for `incremental` build these options don't work

File: tsconfig.json (hunk 2 above)
Comment (splincode) [reply tsconfig.json:40]: I will remove this property

Comment (splincode) [reply tsconfig.json]: fixed

File: package.json (hunk 1 above)
Comment (waterplea) [reply package.json:24]: But this command is just to check for ts errors in precommit. This will increase commit time. Maybe we need a separate tsconfig for it?

Comment (splincode) [reply package.json:24]: fixed

**Review by waterplea — APPROVED** [review]

**Comment by splincode** [comment]
@vladimirpotekhin @nsbarsukov hello, please review me

**Review by vladimirpotekhin — APPROVED** [review]
